src/plugin/plugin_processing_behaviorTriggeredAverage.py

```diff
@@ -0,0 +1,509 @@
+            self.ui.plotLayout.addWidget(self.label_top, stretch=1)
+            self.ui.plotLayout.addWidget(self.label_bot, stretch=2)
+
+        
+        
+    
+    def clearPlotLayout(self):
+        for i in reversed(range(self.ui.plotLayout.count())): 
+            self.ui.plotLayout.itemAt(i).widget().deleteLater()
+    
+    def plotBTA(self):
+        self.clearPlotLayout()
+        self.createAnnotationsImgArray()
+        self.canvas_top = SceneCanvas(size=(self.width,200))
+        self.grid_top = self.canvas_top.central_widget.add_grid()
+        self.xaxis_top = AxisWidget(orientation='bottom',
+                                axis_label='Time (s)',
+                                axis_color='black',
+                                tick_color='black',
+                                text_color='black',
+                                axis_font_size=12,
+                                font_size= 8,
+                                axis_label_margin=16,
+                                tick_label_margin=14)
+        self.xaxis_top.height_max = 35
+        self.xaxis_top.bgcolor = 'white'
+        self.grid_top.add_widget(self.xaxis_top, row=1, col=1)
+
+        self.yaxis_top = AxisWidget(orientation='left',
+                                axis_color='black',
+                                tick_color='black',
+                                text_color='black',
+                                axis_font_size=12,
+                                font_size= 8,
+                                axis_label_margin=16,
+                                tick_label_margin=8)
+        self.yaxis_top.width_max = 35
+        self.yaxis_top.bgcolor = 'white'
+        self.grid_top.add_widget(self.yaxis_top, row=0, col=0)
+
+        self.view_top = self.grid_top.add_view(0, 1, bgcolor='white')
+        self.lineAvgTrials = visuals.Line(
+            np.column_stack((self.trialsTs, self.avgTrials)),
+            parent=self.view_top.scene,
+            color='black'
+        )
+        errPosValues = self.avgTrials + self.errTrials
+        errNegValues = self.avgTrials - self.errTrials
+        if np.count_nonzero(self.errTrials)!=0:
+            notZeroIndices = np.where(~(self.errTrials==0))[0]
+            # handing a case when there are two trials and one has few NaN values
+            errPosValues = self.avgTrials[notZeroIndices] + self.errTrials[notZeroIndices]
+            errNegValues = self.avgTrials[notZeroIndices] - self.errTrials[notZeroIndices]
+            self.errPosLine = visuals.Line(
+                np.column_stack((self.trialsTs[notZeroIndices], errPosValues)),
+                parent=self.view_top.scene,
+                color='white'
+            )
+            self.errNegLine = visuals.Line(
+                np.column_stack((self.trialsTs[notZeroIndices], errNegValues)),
+                parent=self.view_top.scene,
+                color='white'
+            )
+            self.fillBetween = visuals.Polygon(
```

Comment:
I'm getting an error here when I add an offset (making neural data start after behavior):

ls
    self.plotBTA()
  File "K:\Github\bento\src\plugin\plugin_processing_behaviorTriggeredAverage.py", line 385, in plotBTA
    self.fillBetween = visuals.Polygon(
  File "C:\Users\akk7548\Anaconda3\envs\bento\lib\site-packages\vispy\scene\visuals.py", line 131, in __init__
    subclass.__init__(self, *args, **kwargs)
  File "C:\Users\akk7548\Anaconda3\envs\bento\lib\site-packages\vispy\visuals\polygon.py", line 62, in __init__
    self._update()
  File "C:\Users\akk7548\Anaconda3\envs\bento\lib\site-packages\vispy\visuals\polygon.py", line 73, in _update
    pts, tris = data.triangulate()
  File "C:\Users\akk7548\Anaconda3\envs\bento\lib\site-packages\vispy\geometry\polygon.py", line 125, in triangulate
    tri.triangulate()
  File "C:\Users\akk7548\Anaconda3\envs\bento\lib\site-packages\vispy\geometry\triangulation.py", line 186, in triangulate
    self._edge_event(i, j)
  File "C:\Users\akk7548\Anaconda3\envs\bento\lib\site-packages\vispy\geometry\triangulation.py", line 372, in _edge_event
    assert front_index >= 0
    

(possibly this is fixed with the new way of setting time, will check once it's ready!)